### PR TITLE
docs: reorganize documentation

### LINKED ISSUE #1899: docs: improve structure of docs
A new structure has been proposed for the TS SDK documentation.
Move pre-existing documentation to the correct space outlined below:

 (🔴  doesn't currently exist)

---

# Introduction

- Getting Started
    1. Installation
    2. Importing
    3. Using `fuels` from CDN
    4. Connecting to the Network
        1. Connecting to the Testnet
        2. Connecting to a local node
            1. *Crosslink: Tooling utilities*
    5. Checking Wallet Balance (react snippet)
    6. *Crosslink: Querying the Chain (under /providers)*
- 🔴  Creating a Fuel dApp
    1. Shows how to leverage our Tooling and use the `create fuels` CLI  to scaffold a new full-stack dApp
    2. *Show also how to achieve the same manually:*
        1. Write a sample contract
        2. Build your contract *(using `forc`?)*
        3. Running local node *(using `fuel-core`?)*
        4. Deploying contract *(using `forc`?)*
        5. Generating types
        6. Calling contracts
        - *>> May link to the Developer Quickstart?*

# Basics

- Provider
    1. What is it? 
    2. Provider Options?
        1. `fetch`
        2. `timeout`
        3. `cacheUtxo`
        4. `retryOptions` 
        5. `requestMiddleware`
    3. Querying the Chain
        1. Address coins
        2. Address balance
        3. Address spendable resources
        4. Blocks 
    4. Retrying upon errors 
- 🔴  GraphQL
    1. 🔴  Schema?
    2. 🔴  Playground?
    3. 🔴  Custom Queries?
    4. 🔴  Subscriptions?
- Wallet
    1. 🔴  Instantiating Wallet?
    2. Private Keys
        1. Best practices?
    3. Mnemonic Wallet
    4. Encrypting and Decrypting
    5. Checking Balances
    6. Locking & Unlocking
    7. Signing Transactions
    8. 🔴  External Connectors
    9. Test Wallets
        1. Crosslink to: *Generating Test Wallets*
- ABI
    1. 🔴  The JSON ABI file
    2. Generating types
    3. Using generated types

# Essentials

- Contracts
    1. Instantiating a Contract
    2. Calling Contract Functions
        1. Methods *(`get`, `dryuRun`, `simulate`, `call`)*
            1. *Could be just `read` & `submit`*
        2. Transaction Parameters
        3. Call Parameters
        4. Cost Estimation
        5. Dependency estimation
        6. Variable Outputs & Messages
        7. 🔴  Transaction response?
        8. Logs
    3. 🔴  Transaction Policies
    4. 🔴  Custom Transactions
        1. 🔴  Custom Contract call
        2. 🔴  Crosslink Scripts > Custom Script Call
    5. Inter-contract calls
    6. Multiple Contract calls
    7. Using different Wallets
    8. Transferring assets
    9. Deploying Contracts
        1. Storage Slots
        2. Configurable Constants
        3. Understanding the FuelVM Binary File
- Scripts
    1. Instantiating a Script
    2. Configurable Constants
    3. Running scripts
    4. Custom script Call
- Predicates
    1. Instantiating a Predicate
    2. Configurable Constants
    3. Send and Spend funds from Predicate
- Tutorials
    1. Transferring Assets
    2. Deposit & Withdraw
    3. Wallet SDK & React Hooks

# Extras

- Types
    1. Address
        1. Conversion (move standalone page here)
    2. <list of all types, in alphabetical order>
- Errors @ Peter
    1. Error Codes
    2. Revert Errors
- Utilities?
    - Big Number math?
    - Date conversion?
    - Address conversion?
- 🔴  Tutorials?
    1. 🔴  Password Predicate?
    2. 🔴  Predicate swapping tool?

# Tooling

- Testing  @ Nedim
    1. 🔴   Launching nodes
        1.  🔴  Options
        2.  🔴  Examples
    2.  🔴  Launching test nodes
        1.  🔴  Options
        2.  🔴  Examples
    3.  🔴  Generating test wallets
        1.  🔴  Options
        2.  🔴  Examples
    4.  🔴  Test utilities
        1.  🔴  `expectToThrowFuelError()`
        2.  🔴  What else?
- 🔴  CLIs
    1. 🔴  `npm create fuels` CLI
        1. 🔴  Purpose?
        2. 🔴  Options
        3. 🔴  Templates
            1. 🔴  React app
            2. 🔴  NodeJS app?
    2. `fuels` CLI
        1. 🔴  Purpose?
        2. 🔴  Config file
        3. Commands

### PR DESCRIPTION
**Pull request by petertonysmith94**
Closes #1899

Re-organize the menu and page structure for the Docs

---

### Summary

- Seperated our docs into the following sections `Introduction`, `Basics`, `Essentials`, `Extras`, `Tooling`.

Before             |  After
:-------------------------:|:-------------------------:
Screenshot 2024-03-21 at 13 47 45 |  Screenshot 2024-03-21 at 13 47 45

### CODE REVIEW

File: apps/docs/src/extras/utilities/address-conversion.md [hunk 1]
```diff
@@ -1,4 +1,24 @@
-# Converting Between Address Formats and Native Types
+# Address
```
Comment (petertonysmith94) [inline apps/docs/src/extras/utilities/address-conversion.md:1]: Concatenated `guide/address/conversion.md` and `guide/address/index.md` together to get the `address-conversion`

**Comment by petertonysmith94** [comment]
Pipeline currently [failing](https://github.com/FuelLabs/fuels-ts/actions/runs/8365980048/job/22905114913?pr=1906#step:6:1174) on an assertion for the expected structure.

We could update the action [FuelLabs/github-actions](https://github.com/FuelLabs/github-actions/blob/master/docs-hub/vp-docs.js#L15C3-L15C46).

**Comment by danielbate** [comment]
Nice work, I think my preference is option B as I'm concerned about compatibility with the docs hub.

File: apps/docs/src/guide/wallet-manager/index.md
Comment (petertonysmith94) [inline apps/docs/src/guide/wallet-manager/index.md:1]: Consolidated to `apps/docs/src/guide/wallets/wallet-manager.md`

File: apps/docs/src/guide/wallet-manager/locking-and-unlocking-wallet-manager.md
Comment (petertonysmith94) [inline apps/docs/src/guide/wallet-manager/locking-and-unlocking-wallet-manager.md:1]: Consolidated to `apps/docs/src/guide/wallets/wallet-manager.md`

File: apps/docs/src/guide/wallets/wallet-manager.md [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+
+This includes adding new wallets to specific vaults, retrieving all wallets from a vault, exporting specific vaults, and exporting private keys. The `WalletManager` class currently supports two types of vaults: `PrivateKeyVault` and `MnemonicVault`.
+
+### Custom Storage Solutions
+
+The `WalletManager` supports defining a custom storage solution, allowing you to specify how and where the encrypted vaults are saved. With support for custom storage, you can make the `WalletManager` to fit your specific needs and security requirements.
+
+### Locking and Unlocking `WalletManager`
+
+The `WalletManager` implements an automatic encryption mechanism, securely saving the wallet's held vaults. This not only preserves the state of your vaults but also ensures robust protection of the stored information. When needed, you can easily unlock and decrypt the vaults using the previously defined password.
```
Comment (petertonysmith94) [inline apps/docs/src/guide/wallets/wallet-manager.md:17]: This can be simplified IMO

File: apps/docs/src/guide/wallets/wallet-manager.md [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+
 # Getting Started with `WalletManager`
```
Comment (petertonysmith94) [inline apps/docs/src/guide/wallets/wallet-manager.md:19]: Wrong heading indent

File: apps/docs/src/guide/providers/querying-the-chain.md [hunk 4]
```diff
@@ -32,3 +32,23 @@ Get all the spendable balances of all assets for an address. This is different f
...
+
+<<< @/../../../packages/account/src/account.test.ts#Message-getResourcesToSpend{ts:line-numbers}
+
+### Get message proof
+
+A message proof is a cryptographic proof that a message was included in a block. You can use the `getMessageProof` method to retrieve a message proof for a given transaction ID and message ID.
+
+<<< @/../../../packages/account/src/account.test.ts#Message-getMessageProof{ts:line-numbers}
+
+<!-- TODO: Add docs for the two new parameters `commitBlockId` and `commitBlockHeight` -->
```
Comment (petertonysmith94) [inline apps/docs/src/guide/providers/querying-the-chain.md:54]: Consolidated various message docs into here

File: apps/docs/src/guide/predicates/predicate-with-more-complex-args.md
Comment (petertonysmith94) [inline apps/docs/src/guide/predicates/predicate-with-more-complex-args.md:1]: Moved to `apps/docs/src/guide/predicates/instantiating-a-predicate.md`

File: apps/docs/src/guide/predicates/index.md [hunk 5]
```diff
@@ -17,24 +17,6 @@ The key difference here is that instead of checking these rules directly on the
...
-Just like contracts in Sway, once you've created a predicate, you can compile it using `forc build`. For more information on working with Sway, refer to the <a :href="introUrl" target="_blank" rel="noreferrer">Sway documentation</a>.
-
-After compiling, you will obtain the binary of the predicate and its JSON ABI (Application Binary Interface). Using these, you can instantiate a predicate in TypeScript as shown in the code snippet below:
-
-<<< @/../../docs-snippets/src/guide/predicates/index.test.ts#predicate-index-2{ts:line-numbers}
-
-The created [`Predicate`](../../api/Account/Predicate.md) instance, among other things, has three important properties: the predicate `bytes` (byte code), the `chainId`, and the predicate `address`.
-
-This address, generated from the byte code, corresponds to the Pay-to-Script-Hash (P2SH) address used in Bitcoin.
-
```
Comment (petertonysmith94) [inline apps/docs/src/guide/predicates/index.md:37]: Moved to `apps/docs/src/guide/predicates/instantiating-a-predicate.md`

File: apps/demo-fuels/fuels.config.ts [hunk 6]
```diff
@@ -9,5 +9,5 @@ export default createConfig({
 
 /**
  * Check the docs:
- * https://fuellabs.github.io/fuels-ts/guide/cli/config-file
+ * https://fuellabs.github.io/fuels-ts/tooling/cli/fuels/config-file
```
Comment (petertonysmith94) [inline apps/demo-fuels/fuels.config.ts:12]:
```suggestion
 * https://fuellabs.github.io/fuels-ts/guide/fuels/config-file
```

File: apps/docs/src/guide/cli/index.md [hunk 7]
```diff
@@ -1,53 +1,7 @@
...
-
-## Double-checking
-
-```console
-npx fuels -v
-```
-
-## Next Step
-
-Use [`pnpm fuels init`](./commands#init) to create a [`fuel.config.ts`](./config-file) file.
```
Comment (petertonysmith94) [inline apps/docs/src/guide/cli/index.md:53]: Moved to `apps/docs/src/guide/fuels/index.md`

File: apps/docs/src/guide/contracts/index.md [hunk 8]
```diff
@@ -1,14 +1,7 @@
-# Contracts
+# Deploying Contracts
```
Comment (petertonysmith94) [inline apps/docs/src/guide/contracts/index.md:1]:
```suggestion
# Contracts
```

File: apps/demo-fuels/fuels.config.ts (hunk 6 above)
Comment (petertonysmith94) [reply apps/demo-fuels/fuels.config.ts]: https://github.com/FuelLabs/fuels-ts/commit/9a9dbbb9938414dc564426bc19383efde8550e00

File: apps/docs/src/guide/contracts/index.md (hunk 8 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/contracts/index.md]: https://github.com/FuelLabs/fuels-ts/commit/4b47091c942ac4855b9bccc496f0fe220b3a5828

File: apps/docs/src/guide/wallets/wallet-manager.md (hunk 3 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/wallets/wallet-manager.md]: https://github.com/FuelLabs/fuels-ts/commit/2d63f05a58db07b40771289abc9626fbb92f7ccc

File: apps/docs/src/guide/wallets/wallet-manager.md (hunk 2 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/wallets/wallet-manager.md:17]: Will leave for https://github.com/FuelLabs/fuels-ts/issues/1919.

**Review by danielbate — APPROVED** [review]
Great work, good to get this in before we start to evolve the pages.

Approving as this does meet the specification for the restructure.

Couple thoughts:
- Are we keeping glossary, it's not in the new spec? We don't really maintain it and there and the docs hub has a glossary.
- Not sure how I feel about the contents tables, think I preferred the blurb about each section. And also it means we have to maintain 2 orders per section (links json and contents page)

**Review by arboleya — CHANGES_REQUESTED** [review]
Two things only:
 1. Initialize all main sections expanded
 3. Remove the Table of Contents page from all of them
    - Not only because it will be troublesome to maintain, but especially because it will not be present in the docs hub.

**Edit**: And yes, the glossary can go as well.

**Comment by petertonysmith94** [comment]
@arboleya and @danielbate - thanks for reviewing 🙏🏼 

Made the following changed, based on your comments:

- Removed table of contents
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/a9eba896166a597199ef1b78d611f84dd6eeafcf
- Removed glossary
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/653f7f04826cfbb03d39b0fe60fd18b5b26df9bf
- All main sections expanded
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/b1588540ccf8117eccd5d0b88b24750494cdeb82

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**